executor/executor.go

```diff
@@ -262,14 +250,17 @@ func (e *SelectLockExec) Next() (*Row, error) {
 	if row == nil {
 		return nil, nil
 	}
-	if len(row.RowKeys) != 0 && e.Lock == ast.SelectLockForUpdate {
+	if len(e.Schema().TblID2handle) > 0 && e.Lock == ast.SelectLockForUpdate {
```

Comment:
If this map is empty. The following loop will do nothing, so this check is redundant.